File: examples/eApi v1.9/dotNET/CsobGTWDemo/CsobGatewayClientExample/Communication/DataObjects/MallpayOrderItem.cs
using System.Text;
using Newtonsoft.Json;

namespace CsobGatewayClientExample.Communication.DataObjects;

public class MallpayOrderItem : ApiBase
{
    [JsonProperty("code")] public string? Code { get; set; }

    [JsonProperty("ean")] public string? Ean { get; set; }

    [JsonProperty("name")] public string? Name { get; set; }

    [JsonProperty("type")] public string? Type { get; set; }

    [JsonProperty("quantity")] public long? Quantity { get; set; }

    [JsonProperty("variant")] public string? Variant { get; set; }

    [JsonProperty("description")] public string? Description { get; set; }

    [JsonProperty("producer")] public string? Producer { get; set; }

    [JsonProperty("categories")] public List<string>? Categories { get; set; }

    [JsonProperty("unitPrice")] public MallpayPrice? UnitPrice { get; set; }

    [JsonProperty("unitVat")] public MallpayVat? UnitVat { get; set; }

    [JsonProperty("totalPrice")] public MallpayPrice? TotalPrice { get; set; }

    [JsonProperty("totalVat")] public MallpayVat? TotalVat { get; set; }

    [JsonProperty("productUrl")] public string? ProductUrl { get; set; }


    public string ToSign()
    {
        var sb = new StringBuilder();
        Add(sb, Code);
        Add(sb, Ean);
        Add(sb, Name);
        Add(sb, Type);
        Add(sb, Quantity);
        Add(sb, Variant);
        Add(sb, Description);
        Add(sb, Producer);
        if (Categories != null)
            foreach (var category in Categories)
                Add(sb, category);
        if (UnitPrice != null) Add(sb, UnitPrice?.ToSign());
        if (UnitVat != null) Add(sb, UnitVat?.ToSign());
        if (TotalPrice != null) Add(sb, TotalPrice?.ToSign());
        if (TotalVat != null) Add(sb, TotalVat?.ToSign());
        Add(sb, ProductUrl);
        DeleteLast(sb);
        return sb.ToString();
    }
}

File: examples/eApi v1.9/java/eAPI-v19-connector/src/main/java/cz/monetplus/mips/eapi/v19/connector/entity/MallpayOrderItem.java
package cz.monetplus.mips.eapi.v19.connector.entity;

import com.fasterxml.jackson.annotation.JsonInclude;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

import java.util.List;

@Data
@EqualsAndHashCode(callSuper = false)
@JsonInclude(value = JsonInclude.Include.NON_NULL)
@AllArgsConstructor @NoArgsConstructor
public class MallpayOrderItem extends ApiBase implements Signable {
    private String code;
    private String ean;
    private String name;
    private String type;
    private Long quantity;
    private String variant;
    private String description;
    private String producer;
    private List<String> categories;
    private MallpayPrice unitPrice;
    private MallpayVat unitVat;
    private MallpayPrice totalPrice;
    private MallpayVat totalVat;
    private String productUrl;

    @Override
    public String toSign() {
        StringBuilder sb = new StringBuilder();
        add(sb, code);
        add(sb, ean);
        add(sb, name);
        add(sb, type);
        add(sb, quantity);
        add(sb, variant);
        add(sb, description);
        add(sb, producer);
        if(categories != null) for (String category: categories) add(sb, category);
        if(unitPrice != null) add(sb, unitPrice.toSign());
        if(unitVat != null) add(sb, unitVat.toSign());
        if(totalPrice != null) add(sb, totalPrice.toSign());
        if(totalVat != null) add(sb, totalVat.toSign());
        add(sb, productUrl);
        deleteLast(sb);
        return sb.toString();
    }
}
